lib/cretonne/meta/cdsl/registers.py

```diff
@@ -273,12 +272,44 @@ def __getitem__(self, sliced):
         assert sliced.step is None, 'Subclass striding not supported'
 
         w = self.width
-        s = self.start + sliced.start * w
+        s = self.start() + sliced.start * w
```

Comment:
I suppose we get mysterious results if we attempt to slice a non-contiguous register class. I don't think we need to support that, but perhaps an assertion would be in order?

Testing for a contiguous bit mask:

```python
def is_contiguous_mask(x):
    y = x | (x-1)
    return x != 0 and ((y+1) & y) == 0
```

See LLVM's [`isShiftedMask_32` function](https://llvm.org/doxygen/MathExtras_8h_source.html#l00409).

Reply:
Good point. 👍